From a3ed0b4e9f13957e0925bba1f0dfa3881bec82ca Mon Sep 17 00:00:00 2001
From: Todd Rinaldo <toddr@cpanel.net>
Date: Thu, 6 Oct 2016 17:14:53 -0500
Subject: [PATCH 2/2] Do not init PL_strtab if it's already setup

---
 perl.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/perl.c b/perl.c
index 8b1b83c..34f572a 100644
--- a/perl.c
+++ b/perl.c
@@ -299,10 +299,12 @@ perl_construct(pTHXx)
     /* Note that strtab is a rather special HV.  Assumptions are made
        about not iterating on it, and not adding tie magic to it.
        It is properly deallocated in perl_destruct() */
-    PL_strtab = newHV();
+    if(!PL_strtab) {
+        PL_strtab = newHV();
 
-    HvSHAREKEYS_off(PL_strtab);			/* mandatory */
-    hv_ksplit(PL_strtab, 512);
+        HvSHAREKEYS_off(PL_strtab);			/* mandatory */
+        hv_ksplit(PL_strtab, 512);
+    }
 
     Zero(PL_sv_consts, SV_CONSTS_COUNT, SV*);
 
-- 
2.10.0

